server/src/main/java/org/elasticsearch/search/aggregations/metrics/TopHitsAggregator.java

```diff
@@ -86,6 +90,7 @@ private static class Collectors {
         this.subSearchContext = subSearchContext;
         this.topDocsCollectors = new LongObjectPagedHashMap<>(1, bigArrays);
         this.fetchProfiles = context.profiling() ? new ArrayList<>() : null;
+        this.memAccountingBufferSize = context.getClusterSettings().get(SearchService.MEMORY_ACCOUNTING_BUFFER_SIZE).getBytes();
```

Comment:
Another point about the settings, parsing is costly :) I we go for a setting lets store the value outright instead of parsing settings on the hot path?

Reply:
Hm, is this on the hot path - not familiar with aggs but I'd imagine this is at most once / search request?

Reply:
Moved this to the `AggregationContext`